### PR TITLE
Merge extension package into core one

### PR DESCRIPTION
**Pull request by luismiramirez**
The extension package is no longer an independent entity, it is now
merged into @appsignal/nodejs.

The installation script for the extension no longer aborts `npm install`
if something goes wrong during the process. This allows the users to
keep running their apps.

There's no need to use the script that creates the version file as
the extension constants are always available now.
So the script has been removed.

Fixes: #471

### CODE REVIEW

File: packages/nodejs/.changesets/merge-extension-package-into-nodejs.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+bump: "minor"
+type: "change"
+---
+
+The extension package (@appsignal/nodejs-ext) is no longer a separated dependency. Now, it is
+merged into @appsignal/nodejs.
```
Comment (tombruijn) [inline packages/nodejs/.changesets/merge-extension-package-into-nodejs.md:7]:
```suggestion
The extension package (@appsignal/nodejs-ext) is no longer a separated dependency. The extension is part of the @appsignal/nodejs package now.
```

File: packages/nodejs/package.json [hunk 2]
```diff
@@ -26,17 +25,26 @@
     "redis": "*"
   },
   "scripts": {
-    "prebuild": "npm run versionfile",
     "build": "tsc -p tsconfig.json",
-    "build:watch": "npm run versionfile && tsc -p tsconfig.json -w --preserveWatchOutput",
-    "clean": "rimraf dist coverage",
+    "build:ext": "node-gyp rebuild",
```
Comment (tombruijn) [inline packages/nodejs/package.json:29]:
```suggestion
```
I would just remove this. It's like only calling "make" for the Ruby extension installation. That doesn't work if you haven't done all the other steps beforehand already.

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
     "bootstrap:ci": "echo Please run 'mono bootstrap --ci' instead.",
     "build": "echo Please run 'mono build' instead.",
     "build:watch": "echo Please run 'mono run npm run build:watch --parallel' instead.",
-    "build:ext": "echo Please run 'mono run npm run build:ext --package @appsignal/nodejs-ext' instead.",
+    "build:ext": "echo Please run 'mono run npm run build:ext --package @appsignal/nodejs' instead.",
```
Comment (tombruijn) [inline package.json:31]:
```suggestion
```
See my other comment about this script.

File: packages/nodejs/package.json [hunk 4]
```diff
@@ -26,17 +25,26 @@
...
+    "test:watch": "jest --filter=./test/filter.js --watch"
   },
+  "os": [
+    "linux",
+    "darwin",
+    "freebsd"
+  ],
+  "cpu": [
+    "x64",
+    "x86"
```
Comment (tombruijn) [inline packages/nodejs/package.json:46]: Do we need to include ARM here? We support that for Node.js too right?

File: packages/nodejs/scripts/extension/extension.js [hunk 5]
```diff
@@ -6,13 +6,13 @@ const fs = require("fs")
 const crypto = require("crypto")
 const childProcess = require("child_process")
 
-const { AGENT_VERSION, MIRRORS, TRIPLES } = require("./extension/constants")
+const { AGENT_VERSION, MIRRORS, TRIPLES } = require("./support/constants")
```
Comment (tombruijn) [inline packages/nodejs/scripts/extension/extension.js:9]: Can you also send in a PR for the agent repo to update this path?

File: packages/nodejs/src/diagnose.ts [hunk 6]
```diff
@@ -265,7 +265,7 @@ export class DiagnoseTool {
   }
 }
 
-// This implementation should match the `packages/nodejs-ext/scripts/report.js`
+// This implementation should match the `/scripts/extension/report.js`
```
Comment (tombruijn) [inline packages/nodejs/src/diagnose.ts:268]:
```suggestion
// This implementation should match the `scripts/extension/report.js`
```
The leading `/` implies root of the file system.

**Review by tombruijn — CHANGES_REQUESTED** [review]

File: packages/nodejs/scripts/extension/extension.js (hunk 5 above)
Comment (luismiramirez) [reply packages/nodejs/scripts/extension/extension.js:9]: PR: https://github.com/appsignal/appsignal-agent/pull/738

**Review by tombruijn — APPROVED** [review]

**Review by jeffkreeftmeijer — APPROVED** [review]

**Pull request merged**